x/group/simulation/operations.go

```diff
@@ -699,6 +713,133 @@ func SimulateMsgUpdateGroupPolicyMetadata(ak group.AccountKeeper,
 	}
 }
 
+// SimulateMsgWithdrawProposal generates a MsgWithdrawProposal with random values
+func SimulateMsgWithdrawProposal(ak group.AccountKeeper,
+	bk group.BankKeeper, k keeper.Keeper) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, sdkCtx sdk.Context, accounts []simtypes.Account, chainID string) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		g, groupPolicy, _, _, err := randomGroupPolicy(r, k, ak, sdkCtx, accounts)
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, ""), nil, err
+		}
+		if g == nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "no group found"), nil, nil
+		}
+		if groupPolicy == nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "no group policy found"), nil, nil
+		}
+		groupPolicyAddr := groupPolicy.Address
+
+		// Pick a random member from the group
+		ctx := sdk.WrapSDKContext(sdkCtx)
+		acc, account, err := randomMember(r, k, ak, ctx, accounts, g.GroupId)
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, ""), nil, err
+		}
+		if account == nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "no group member found"), nil, nil
+		}
+
+		spendableCoins := bk.SpendableCoins(sdkCtx, account.GetAddress())
+		fees, err := simtypes.RandomFees(r, sdkCtx, spendableCoins)
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, "fee error"), nil, err
+		}
+
+		policy := groupPolicy.GetDecisionPolicy()
+		err = policy.Validate(*g)
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, TypeMsgWithdrawProposal, err.Error()), nil, nil
+		}
+
+		proposalReq, err := group.NewMsgCreateProposalRequest(groupPolicyAddr, []string{acc.Address.String()}, []sdk.Msg{
+			&banktypes.MsgSend{
+				FromAddress: groupPolicyAddr,
+				ToAddress:   accounts[1].Address.String(),
+				Amount:      sdk.Coins{sdk.NewInt64Coin("token", 100)},
+			},
+		}, []byte{}, 0)
```

Comment:
why are you creating a new proposal here?

Reply:
in `TestWeightedOperations` test the proposal which is created in `SimMsgCreateProposal` is closing in `SimMsgExec`, This test is returning NoOpMsg since there are no open proposals found.

Reply:
then what about handling that in the test `TestWeightedOperations` and `WeightedOperations` instead?
see for instance how it's done in `x/gov/simulation/operations.go` `WeightedOperations`